### DISCUSSION TITLE
[@types/stripe-v3] outdated types

### DISCUSSION THREAD

**Discussion by epelc** [DefinitelyTyped/DefinitelyTyped#59227] [Issues with a @types package]
Hello,

The types for @types/stripe-v3 which tracks the stripe.js api are outdated. 

For example they are missing the appearance field on `stripe.elements({ appearance, clientSecret})`. It appears their loader library has updated types but they aren't published here.

https://github.com/stripe/stripe-js

[a comment by an automated account is not shown]

**Comment by kaznovac** [comment]
`@types/stripe-v3` should probably not be used as https://github.com/stripe/stripe-js has/provides native types support.

the aforementioned should probably get removed from DT (consult the readme section about [making a contribution](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#how-can-i-contribute), or more concisely [removing a package](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#removing-a-package))

however `@types/stripe-v3` is used in `@types/react-stripe-elements` the type definitions for https://github.com/stripe-archive/react-stripe-elements (an archived project moved to https://github.com/stripe/react-stripe-js with a native TS support)

these are the paths (i see):
- remove both types (a bit controversial, `@types\react-stripe-elements` have ~1e5 downloads on npmjs)
- remove `@types\stripe-v3` and adapt `@types\react-stripe-elements` to use official types (a bit mundane, updating types for a replaced library that provides it's own types)
- let it be (is it more confusing?)

this is a bit of a tricky decision, most possibly for the DT maintainers to address

**Comment by epelc** [comment]
@kaznovac perhaps could just have `@types/stripe-v3` mention the existence of `@stripe/stripe-js` and that it may have more up to date type/info?

Would be nice if someone from stripe could chime in on the actual status here. Just pinging a couple of recent committers on `@stripe/stripe-js` to see if they can add some info. @jackieosborn-stripe @christopher-stripe @alic-stripe


We have actually switched to using `@stripe/stripe-js` itself due to the types for elements being outdated. It does come with a loader but we really just wanted raw types since the loader is a bit overkill code size wise for us.

**Comment by christopher-stripe** [comment]
Hey @epelc, we (Stripe) do maintain TypeScript definitions for Stripe.js in the `@stripe/stripe-js` package now and plan to keep doing so. I don't know the status of these DefinitelyTyped definitions—as far as I know they have always been a community-run project.

> It does come with a loader but we really just wanted raw types since the loader is a bit overkill code size wise for us.

I bet you figured this out already, but in case anyone else is wondering you can use TypeScript's [type-only imports](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html#type-only-imports-and-export) if you don't want the loader behavior.

```ts
import type {Stripe} from '@stripe/stripe-js'
```

**Comment by epelc** [comment]
Thanks @christopher-stripe! I wasn't aware of the `import type` syntax it is exactly what I need.
